### PR TITLE
feat: attach masks on projectile hit and adjust behavior on antis

### PR DESCRIPTION
**Pull request by LouisAndrew**
### Description

- Attach masks when hitting NPC (OP and FFP2)
- When FFP2 is equipped, shooting OP mask to NPC won't do a thing
- Set behavior of NPC to `Cured` when hit by projectile from `Syringe` weapon type
- Add tags to projectiles (see `Weapons.WeaponTags`)
- Adjusted behavior on antis -> Syringe won't work on antivac, masks won't be attached to antimasks
- Add additional attributes to `NPCSpawner` for testing (and maybe for further use e.g. spawning antivacs)

### How to Test
Scene: `PlayerHealthIntegration`
- Shoot different projectiles on different NPCs

### Bugs
- FFP2 positioning on female NPC seems weird

### Visual Support
Demo Link: [https://drive.google.com/file/d/11CkyTUYuN9YtreYYqEkBy-8OgRkl5cLo/view?usp=sharing](https://drive.google.com/file/d/11CkyTUYuN9YtreYYqEkBy-8OgRkl5cLo/view?usp=sharing)

### CODE REVIEW

**Review by Krysenkova — APPROVED** [review]
On my side everything works perfectly

File: Assets/Scripts/NPC/NPCBehavior.cs [hunk 1]
```diff
@@ -35,12 +54,16 @@ public class NPCBehavior : MonoBehaviour
         private NPCWaveManager _waveManager;
         private Animator _animator;
 
+        private int _isMasked; // between 0 and 2, 0 = nomask, 1 = op mask, 2 = ffp mask
```
Comment (hunsri) [inline Assets/Scripts/NPC/NPCBehavior.cs]: I think a MaskType enum would be better here

**Review by hunsri — CHANGES_REQUESTED** [review]

**Comment by LouisAndrew** [comment]
Sounds are a bit weird when there are a lot of infected on the same place at the same time

**Comment by LouisAndrew** [comment]
Duplicated animated NPCs with accessories to avoid conflict with #38

File: Assets/Scripts/NPC/NPCBehavior.cs [hunk 2]
```diff
@@ -215,5 +288,16 @@ public void triggerInfectedSoundsInInterval()
                 SetTimeInterval();
             }
         }
+
+        /// <summary>
+        /// For debug purposes.
+        /// </summary>
+        public void SetBehaviors(bool isNoMask, bool isNoVac, MaskType maskType)
```
Comment (hunsri) [inline Assets/Scripts/NPC/NPCBehavior.cs:295]: Is that intended for debug only?

**Review by hunsri — APPROVED** [review]
Should be ok now :+1:

File: Assets/Scripts/NPC/NPCBehavior.cs (hunk 2 above)
Comment (LouisAndrew) [reply Assets/Scripts/NPC/NPCBehavior.cs:295]: it's used later for the settings. Will change the comment on future commit

**Pull request merged**